mito-ai/mito_ai/completions/handlers.py

```diff
@@ -222,7 +222,25 @@ async def on_message(self, message: str) -> None: # type: ignore
                 )
                 self.reply(reply)
             return
-        
+
+        if type == MessageType.STOP_AGENT:
+            # Get the thread_id from metadata if available, otherwise use the newest thread
+            thread_id_to_stop = metadata_dict.get('threadId')
+
+            if not thread_id_to_stop:
+                # Get the newest thread ID
+                thread_id_to_stop = message_history._get_newest_thread_id()
+            
+            if thread_id_to_stop:
+                self.log.info(f"Stopping agent, thread ID: {thread_id_to_stop}")
+                message_history.remove_last_message(
```

Comment:
Instead of calling `remove_last_message` let's call a function like: `add_thread_id_to_ignore_list`. 

Then, we just have to update the `message_history.append_message()` function so that it first checks the `thread_id_to_ignore_list`.

If the thread_is is present, 1) don't append the message 2) remove the thread_id from the list 3) and return. 

If the thread_is is not present: continue as we normally do.